src/Components/MelodyEditor/MelodyElement/index.jsx

```diff
@@ -34,15 +34,18 @@ const MelodyElement = forwardRef(({
   const [isAccepted, setIsAccepted] = useState(accepted);
   const [isPlayable, setIsPlayable] = useState(false);
   const [playing, setPlaying] = useState(false);
-  const stop = useRef(false);
+  const oscillator = useRef(null);
   const highlighted = useRef(null);
 
   useImperativeHandle(ref, () => ({
     play() {
       playMelody();
     },
     stop() {
-      stop.current = true;
+      if (oscillator.current) {
```

Comment:
I suppose I should have just called stopMelody() here...

Reply:
Yep.